### DISCUSSION TITLE
Audit

### DISCUSSION THREAD

**Discussion by enenum** [pomerium/pomerium#2022] [Q&A] [answered]
Hi there. Awesome work you guys!!! Keep it up.

I am seeking to understand if pomerium offers the ability to audit the request that come through the proxy. Seeing it is in a central place that grants it a golden opportunity, nad the mention of audit in the code and documentation, there is a great silence about how to use such a feature if indeed it is functional.

If it is in the works, it would be good to hear what the plans are - it might help in choosing the pomerium for the stack.

**Comment by calebdoxsey** [comment, marked as answer]
Hi @enenum , we have basic logging for each authorization request. This includes http request information (path, method, etc) as well as session and user information.

We have plans to support more exhausting auditing capabilities, some of which I'm currently working on (the tricky thing is that there are parts of the request which are quite sensitive and need to be encrypted since logs have a tendency to leak in the wild).

Do you have more details about what you were looking for?

**Reply by enenum** [reply]
Thank you @calebdoxsey. I think the status answer is good for now. If I get more insights, I will update. And you do bring up an important issue - the security of log info. Are you planning to enable it separately from normal logging or to output it to a separate stream so it can be easily directed to an encrypted destination, or mix it with the normal logging?

**Reply by calebdoxsey** [reply]
As of now our plan is to have it in the same log stream as normal logs. The message itself is encrypted so is safe for general consumption.

In previous discussions we've found most people have some sort of log stack already, and most of those tools should be able to specify rules for how to route logs. We will build out some tooling to help with decrypting the logs.

We are also working on some enterprise features that will make auditing easier to use. If you're interested feel free to reach out to @desimone . Much of this work is quite early and we welcome suggestions. Thanks.
